### PR TITLE
refactor: add N_m3u8DL-RE

### PR DESCRIPTION
**Pull request by candrapersada**
# Pull Request Template

## Type of change

- [ ] Bug fix
- [x] Feature
- [ ] Documentation update

## Description

*ramble here*

## Checklist

- [ ] any anime playing
- [ ] bumped version
---
- [ ] next, prev and replay work
- [ ] `-c` history and continue work
- [ ] `-d` downloads work
- [ ] `-s` syncplay works
- [ ] `-q` quality works
- [ ] `-v` vlc works
- [ ] `-e` select episode works
- [ ] `-S` select index works
- [ ] `-r` range selection works
- [ ] `--dub` and regular (sub) mode both work
- [ ] all providers return links (not necessarily on a single anime, use debug mode to confirm)
---
- [ ] `-h` help info is up to date
- [ ] Readme is up to date
- [ ] Man page is up to date

## Additional Testcases

- The safe bet: One Piece
- Episode 0: Saenai Heroine no Sodatekata ♭
- Unicode: Saenai Heroine no Sodatekata ♭
- Non-whole episodes: Tensei shitara slime datta ken (ep. 24.5, ep. 24.9)

### CODE REVIEW

**Review by justchokingaround — COMMENTED** [review]
wtf is N_m3u8DL

**Comment by candrapersada** [comment]
This is an m3u8 downloader.

https://github.com/nilaoda/N_m3u8DL-CLI/blob/master/README_ENG.md

**Comment by justchokingaround** [comment]
is it faster than yt-dlp?

**Comment by port19x** [comment]
Now that is an obscure patch!

I'll look into that m3u8dl accelerator

**Comment by port19x** [comment]
Since it's just a +2 line diff I'm not opposed to including it

**Comment by port19x** [comment]
@Derisis13 your call m8

File: ani-cli [hunk 1]
```diff
@@ -224,7 +224,9 @@ update_history() {
 download() {
     case $1 in
         *m3u8*)
-            if command -v "yt-dlp" >/dev/null; then
+            if command -v "N_m3u8DL-RE" >/dev/null; then
+                N_m3u8DL-RE --log-level ERROR --thread-count 10 --download-retry-count 10 "$1" --save-dir "$download_dir" --save-name "$2"
```
Comment (Derisis13) [inline ani-cli:228]: Is there any particular reason to run it with less threads than yt-dlp? (10 as oppose to 16)
Was there a process in selecting these two numbers or are they just rules of thumb?

Comment (candrapersada) [reply ani-cli:228]: this is only temporary until approved add N_m3u8DL-RE

**Comment by port19x** [comment]
Nah

**Pull request closed without merging**